qutip/core/qobj.py

```diff
@@ -1055,7 +1032,7 @@ def unit(self, inplace=False, norm=None, kwargs=None):
         """
         norm = self.norm(norm=norm, kwargs=kwargs)
         if inplace:
-            self.data /= norm
+            _data.imul(self.data, 1/norm)
```

Comment:
This is unrelated to the PR - probably best not include it.  It's also not necessarily correct; for example, it'll make data backends with autodifferentiation slower/less accurate, because you're masking the true operation.  It should probably be up to the data backends to decide whether this is an acceptable modification.

Also, this is a good illustration of my reservations about inplace operations on the data layer; we said that they're not guaranteed to be in-place, so you have to assign the result.